commp.go

```diff
@@ -50,42 +50,49 @@ const MaxPiecePayload = MaxPieceSize / 128 * 127
 // at least this amount of bytes.
 const MinPiecePayload = uint64(65)
 
+const (
+	commpDigestSize = sha256simd.Size
+	quadPayload     = 127
+	bufferSize      = 256 * quadPayload // FIXME: tune better, chosen by rough experiment
+)
+
 var (
-	layerQueueDepth   = 256 // SANCHECK: too much? too little? can't think this through right now...
+	layerQueueDepth   = 32 // FIXME: tune better, chosen by rough experiment
 	shaPool           = sync.Pool{New: func() interface{} { return sha256simd.New() }}
 	stackedNulPadding [MaxLayers][]byte
 )
 
 // initialize the nul padding stack (cheap to do upfront, just MaxLayers loops)
 func init() {
 	h := shaPool.Get().(hash.Hash)
-	defer shaPool.Put(h)
 
-	stackedNulPadding[0] = make([]byte, 32)
+	stackedNulPadding[0] = make([]byte, commpDigestSize)
```

Comment:
// not a review, just capturing for posterity

`sha256simd.Size` is also 32 -- https://github.com/minio/sha256-simd/blob/v1.0.0/sha256.go#L29